src/core/state/networks/networks.ts

```diff
@@ -198,10 +198,10 @@ function createSelector<T>(
 function createParameterizedSelector<T, Args extends unknown[]>(
   selectorFn: (params: {
     networks: Networks;
-    userPreferences: Record<number, ChainPreferences>;
+    userPreferences: Record<number, ChainPreferences | undefined>;
     chainOrder: number[];
     enabledChainIds: Set<number>;
-    mergedChainData: Record<number, TransformedChain>;
+    mergedChainData: Record<number, TransformedChain | undefined>;
```

Comment:
Was there a reason we needed undefined here?

Reply:
in a record, undefined is always a possibility :)
bth I'm not sure why typescript does not do this by default.
There's no Record (object, map) where each key is defined, unless you work with getters